tests/snappi/pfc/test_pfc_pause_response_with_snappi.py

```diff
@@ -0,0 +1,163 @@
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [ pytest.mark.topology('tgen') ]
+
+def test_pfc_single_lossless_headroom(snappi_api,
+                                        snappi_testbed_config,
+                                        conn_graph_facts,
+                                        fanout_graph_facts,
+                                        duthosts,
+                                        rand_one_dut_hostname,
+                                        rand_one_dut_portname_oper_up,
+                                        enum_dut_lossless_prio,
+                                        all_prio_list,
+                                        prio_dscp_map,
+                                        pfc_pause_delay_test_params):
+    """
+    Test headroom capacity for DUT for a single lossless priority
+
+    Args:
+        snappi_api (pytest fixture): SNAPPI session
+        snappi_testbed_config (pytest fixture): testbed configuration information
+        conn_graph_facts (pytest fixture): connection graph
+        fanout_graph_facts (pytest fixture): fanout graph
+        duthosts (pytest fixture): list of DUTs
+        rand_one_dut_hostname (str): hostname of DUT
+        rand_one_dut_portname_oper_up (str): port to test, e.g., 's6100-1|Ethernet0'
+        enum_dut_lossless_prio (str): lossless priority to test, e.g., 's6100-1|3'
+        all_prio_list (pytest fixture): list of all the priorities
+        prio_dscp_map (pytest fixture): priority vs. DSCP map (key = priority).
+        pfc_pause_delay_test_params (pytest fixture): dictionary of pfc pause delay values, 
+                                                  and delay responses e.g. {1:True, 2:False}
+
+    Returns:
+        N/A
+    """
+
+    pytest_require(pfc_pause_delay_test_params is not None, 
+                    "Skip this testcase since pfc pause delay values have not been configured yet")
+    dut_hostname, dut_port = rand_one_dut_portname_oper_up.split('|')
+    dut_hostname2, lossless_prio = enum_dut_lossless_prio.split('|')
+    pytest_require(rand_one_dut_hostname == dut_hostname == dut_hostname2,
+                   "Priority and port are not mapped to the expected DUT")
+
+    testbed_config, port_config_list = snappi_testbed_config
+    duthost = duthosts[rand_one_dut_hostname]
+    lossless_prio = int(lossless_prio)
+
+    pause_prio_list = [lossless_prio]
+    test_prio_list = [lossless_prio]
+    bg_prio_list = [p for p in all_prio_list]
+    bg_prio_list.remove(lossless_prio)
+
+    """ Modify pfc pause delay value """
+    l1_config = testbed_config.layer1[0]
+    pytest_require(len(l1_config.port_names) > 1,
+                   "Skip this testcase since only one Rx and Tx port have been configured")
+    pfc = l1_config.flow_control.ieee_802_1qbb
+
+    for pfc_pause_delay, headroom_test_result in pfc_pause_delay_test_params.items():
```

Comment:
Instead of having a for loop here, it seems to me that we should have an enum for pfc_pause_delay_test_params